src/plugins/graph/components/graph-axis.tsx

```diff
@@ -52,6 +56,38 @@ export const GraphAxis = observer(function GraphAxis(
     droppedAttrId && isDropAllowed(place, droppedAttrId) && onDropAttribute?.(place, droppedAttrId);
   });
 
+  useEffect(() => {
+    if (autoAdjust?.current) {
+      dataConfig?.onAction(action => {
+        if (
+            isAlive(graphModel) &&
+            (isAddCasesAction(action) || isSetCaseValuesAction(action))
+           )
+        {
+          const xValues = dataConfig.numericValuesForAttrRole("x");
+          const yValues = dataConfig.numericValuesForAttrRole("y");
+          const axisModel = graphModel.getAxis(place);
+
+          if (axisModel && isNumericAxisModel(axisModel)) {
+            if (xValues && place === "bottom") {
+              const minX = Math.min(...xValues);
+              const maxX = Math.max(...xValues);
+              const newXBounds = computeNiceNumericBounds(minX, maxX);
```

Comment:
I should have thought of this when we were working on it earlier, but does the PT story ask for auto-adjust or auto-expand, i.e. are the axes supposed to shrink as well as grow? I thought it was just supposed to auto-expand if any values are not visible within the current bounds. This code will reset user-set bounds even if all of the values are visible. I think the right thing to do is to compare the min and max to the current axis bounds and only change the bounds if necessary.

Reply:
Sorry, looks like I conflated stories and their requirements. It's actually [this story](https://www.pivotaltracker.com/story/show/184187726) that spells out more how things should synch. So I don't think we have to nail this down perfectly for this PR.

I made the change you suggested, I think. However, that other story says:

"If the domain/range of the data is changed the plot will update to match (note that Geometry tile does this now)."

I checked out how the Geometry tile works, and it does behave how the Graph's auto-expand worked before the very latest change. So the axes will auto shrink in a Geometry tile. However, the Geometry tile doesn't seem to have the option to manually adjust the axes, so I'm not sure we want the two to behave exactly the same. I can ask Leslie about that, but I don't think we have to wait for the answer to finish up this PR. I think we can finalize that when I do the work for that other story about synchronization. Does that seem reasonable?

Reply:
@kswenson Per Leslie's comment [on PT ](https://www.pivotaltracker.com/story/show/184187726/comments/236729743), I've undone the previous commit with the above-mentioned changes. So now the aces will always auto-adjust.